hera_pspec/tests/test_grouping.py

```diff
@@ -718,5 +718,39 @@ def test_spherical_wf_from_uvp():
                   little_h='h^-3' in uvp.norm_units)
 
 
-if __name__ == "__main__":
-    unittest.main()
+class TestAverageDelayBins:
+    """Tests of the average_in_delay_bins function."""
+    
+    def setup_class(self):
+        beamfile = os.path.join(DATA_PATH, 'HERA_NF_dipole_power.beamfits')
+        self.beam = pspecbeam.PSpecBeamUV(beamfile)
+        uvp, cosmo = testing.build_vanilla_uvpspec(beam=self.beam)
+        self.uvp = uvp
+        
+        self.uvp_nocov = copy.deepcopy(self.uvp)
+        del self.uvp_nocov.cov_array_real
+        del self.uvp_nocov.cov_array_imag
+        
```

Comment:
Maybe add a test with exact window functions? or raise an error to say they're not supported?
Test could be with object:
`uvp2 = copy.deepcopy(self.uvp)
gaussian_beam = uvwindow.FTBeam.gaussian(freq_array=uvp2.freq_array[0], widths=8., pol=1) 
 uvp2.get_exact_window_functions(ftbeam=gaussian_beam, inplace=True)`